src/hotspot/share/cds/finalImageRecipes.cpp

```diff
@@ -127,6 +127,14 @@ void FinalImageRecipes::record_recipes_for_constantpool() {
     }
 
     if (cp_indices.length() > 0) {
+      LogStreamHandle(Trace, aot, resolve) log;
```

Comment:
This is very helpful information However, am I right in thinking this log output is going to be emitted during the training run?

It would really help if we could log details of which CP entries are resolved during an assembly run and/or a production run so we could associate linkage info, including any referenced metadata and/or heap roots, with the corresponding CP/CPCache/Klass. That would allow the map analysis to be extended to show how assets in the cache are connected to their owning Klass(es).